components/shopify/sources/updated-customer/updated-customer.js

```diff
@@ -0,0 +1,36 @@
+const shopify = require("../../shopify.app.js");
+
+module.exports = {
+  key: "shopify-updated-customer",
+  name: "Updated Customer",
+  description: "Emits an event each time a customer's information is updated.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    shopify,
+  },
+  async run() {
+    const lastUpdatedAt = this.db.get("last_updated_at") || null;
+    let results = await this.shopify.getCustomers(null, lastUpdatedAt);
+
+    for (const customer of results) {
+      const dedupeId = `${customer.id}-${customer.updated_at}`;
+      this.$emit(customer, {
+        id: dedupeId,
+        summary: `${customer.first_name} ${customer.last_name}`,
+        ts: Date.now(),
```

Comment:
For this field, `customer.updated_at` would be a more appropiate value.
We use `Date.now()` as a last resort for a timestamp since it's the less likely to reflect the time at which an event happened in the original source.

In order of relevance, these are the values that you can use for this field:
- `customer.updated_at` (use `Date.parse` so that you get a JS timestamp)
- `event.timestamp` (x1000 since this value is expressed in seconds, not milliseconds like JS timestamps)
- `Date.now()`